Source/WebKit/Platform/IPC/IPCSemaphore.h

```diff
@@ -85,4 +94,93 @@ class Semaphore {
 #endif
 };
 
+// FIXME: Write proper interruptible implementations for non-COCOA platforms.
+class InterruptibleBinarySemaphoreSignaler {
```

Comment:
I'd still prefer IPC::Event in (IPCEvent.h) on the count of this being quite a mouthful.
With the documentation that it's interruptible.